src/payment/strategies/affirm/affirm-payment-strategy.ts

```diff
@@ -256,27 +253,44 @@ export default class AffirmPaymentStrategy implements PaymentStrategy {
 
     private _getDiscounts(): AffirmDiscount {
         const state = this._store.getState();
-        const cart = state.cart.getCart();
+        const order = state.order.getOrder();
+        const transformer = this._getTransformer();
 
-        if (!cart) {
-            throw new MissingDataError(MissingDataErrorType.MissingCart);
+        if (!order) {
+            throw new MissingDataError(MissingDataErrorType.MissingCheckout);
         }
 
         const discounts: AffirmDiscount = {};
-        for (const line of cart.coupons) {
-            discounts[line.code] = {
-                discount_amount: line.discountedAmount,
-                discount_display_name: line.displayName,
-            };
+
+        for (const line of order.coupons) {
+            if (line.discountedAmount > 0) {
+                discounts[line.code] = {
+                    discount_amount: transformer.toInteger(line.discountedAmount),
+                    discount_display_name: line.displayName,
+                };
+            }
         }
-        for (const line of cart.discounts) {
-            discounts[line.id] = {
-                discount_amount: line.discountedAmount,
-                discount_display_name: line.id,
+
+        if (order.discountAmount > 0) {
+            discounts.discounts = {
+                discount_amount: transformer.toInteger(order.discountAmount),
+                discount_display_name: 'discount',
             };
         }
 
         return discounts;
     }
 
+    private _getTransformer(): AmountTransformer {
```

Comment:
I think the AmountTransformer should be initialized at the constructor, or received as a constructor parameter, what do you think @davidchin ?